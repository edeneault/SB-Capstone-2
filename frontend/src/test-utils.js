import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
