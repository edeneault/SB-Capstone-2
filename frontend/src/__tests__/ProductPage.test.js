import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { render } from "../test-utils";
import { initialState } from "../store-mock";
import ProductPage from "../pages/ProductPage.js";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  test("renders ProductPage without crashing", async () => {
    const store = mockStore(initialState);
    const match = {
      params: {
        keyword: "",
      },
    };

    const { getByText } = render(
      <Provider store={store}>
        <ProductPage match={match} />
      </Provider>,
      { initialRoutes: [`/allproducts`] },
    );

    expect(getByText(/Add to cart/i)).toBeInTheDocument();
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);
    const match = {
      params: {
        keyword: "",
      },
    };

    const { asFragment } = render(
      <Provider store={store}>
        <ProductPage match={match} />
      </Provider>,
      { initialRoutes: [`/allproducts`] },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
