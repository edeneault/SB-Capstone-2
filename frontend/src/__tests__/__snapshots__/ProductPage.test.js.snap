// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="px-5 container-fluid"
  >
    <button
      class="btn btn-light my-3 btn btn-primary"
      type="button"
    >
      Go Back
    </button>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <img
          alt="Lavazza - Espresso - Medium Roast"
          class="fade-in img-fluid"
          src="https://res.cloudinary.com/edeneault-cloudinary/image/upload/v1630783263/eShop/sqiwu5ummm4y1to3cp5o.jpg"
        />
      </div>
      <div
        class="col-md-3"
      >
        <div
          class="list-group list-group-flush"
        >
          <div
            class="list-group-item"
          >
            <h3>
              Lavazza - Espresso - Medium Roast
            </h3>
          </div>
          <div
            class="list-group-item"
          >
            <div
              class="rating"
            >
              <span>
                <i
                  class="fas fa-star"
                  style="color: rgb(140, 61, 61);"
                />
              </span>
              <span>
                <i
                  class="fas fa-star"
                  style="color: rgb(140, 61, 61);"
                />
              </span>
              <span>
                <i
                  class="fas fa-star"
                  style="color: rgb(140, 61, 61);"
                />
              </span>
              <span>
                <i
                  class="fas fa-star-half-alt"
                  style="color: rgb(140, 61, 61);"
                />
              </span>
              <span>
                <i
                  class="far fa-star"
                  style="color: rgb(140, 61, 61);"
                />
              </span>
              <span>
                 10  reviews
              </span>
            </div>
          </div>
          <div
            class="list-group-item"
          >
            Price: $9.99
          </div>
        </div>
      </div>
      <div
        class="col-md-3"
      >
        <div
          class="rounded card"
        >
          <div
            class="list-group list-group-flush"
          >
            <div
              class="list-group-item"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Price:
                </div>
                <div
                  class="col"
                >
                  <strong>
                    $9.99
                  </strong>
                </div>
              </div>
            </div>
            <div
              class="list-group-item"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Status:
                </div>
                <div
                  class="col"
                >
                  In Stock
                </div>
              </div>
            </div>
            <div
              class="list-group-item"
            >
              <div
                class="row"
              >
                <div
                  class="col"
                >
                  Qantity:
                </div>
                <div
                  class="col"
                >
                  <form
                    class=""
                  >
                    <select
                      class="form-control"
                    >
                      <option
                        value="1"
                      >
                        1
                      </option>
                      <option
                        value="2"
                      >
                        2
                      </option>
                      <option
                        value="3"
                      >
                        3
                      </option>
                      <option
                        value="4"
                      >
                        4
                      </option>
                    </select>
                  </form>
                </div>
              </div>
            </div>
            <div
              class="list-group-item"
            >
              <div
                class="d-grid gap-2 col"
              >
                <button
                  class="btn-block rounded btn btn-primary"
                  type="button"
                >
                  Add To Cart
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="me-3  row"
    >
      <div
        class="rounded p-3 m-3 card"
        style="min-height: 50px;"
      >
        Description: Lavazza - 2.2 lbs (1kg) Whole Bean - Medium Roast
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <h2>
          Reviews
        </h2>
        <div
          class="fade rounded alert alert-info show"
          role="alert"
        >
          No Reviews
        </div>
        <div
          class="list-group list-group-flush"
        />
      </div>
      <div
        class="col-md-6"
      >
        <div
          class="list-group list-group-flush"
        >
          <div
            class="list-group-item"
          >
            <h2>
              Write a Customer Review
            </h2>
            <form
              class=""
            >
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="rating"
                >
                  Rating
                </label>
                <select
                  class="form-control"
                  id="rating"
                >
                  <option
                    value=""
                  >
                    Select...
                  </option>
                  <option
                    value="1"
                  >
                    1 - Poor
                  </option>
                  <option
                    value="2"
                  >
                    2 - Fair
                  </option>
                  <option
                    value="3"
                  >
                    3 - Good
                  </option>
                  <option
                    value="4"
                  >
                    4 - Very Good
                  </option>
                  <option
                    value="5"
                  >
                    5 - Excellent
                  </option>
                </select>
              </div>
              <div
                class="form-group"
              >
                <label
                  class="form-label"
                  for="comment"
                >
                  Comment
                </label>
                <textarea
                  class="form-control"
                  id="comment"
                  row="3"
                />
              </div>
              <button
                class="my-3 rounded btn btn-primary"
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
